pkg/apis/hub/v1alpha1/api_auth.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright (C) 2022-2025 Traefik Labs
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published
+by the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// APIAuth defines the authentication configuration for APIs.
+type APIAuth struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired behavior of this APIAuth.
+	Spec APIAuthSpec `json:"spec,omitempty"`
+
+	// The current status of this APIAuth.
+	// +optional
+	Status APIAuthStatus `json:"status,omitempty"`
+}
+
+// APIAuthSpec configures the authentication for APIs.
+// +kubebuilder:validation:XValidation:message="exactly one of apiKey or jwt must be specified",rule="(has(self.apiKey) && !has(self.jwt)) || (!has(self.apiKey) && has(self.jwt))"
+type APIAuthSpec struct {
+	// IsDefault specifies if this APIAuth should be used as the default API authentication method for the namespace.
+	// Only one APIAuth per namespace should have isDefault set to true.
+	IsDefault bool `json:"isDefault"`
+
+	// APIKey configures API key authentication.
+	// +optional
+	APIKey *APIKeyAuthSpec `json:"apiKey,omitempty"`
+
+	// JWT configures JWT authentication.
+	// +optional
+	JWT *JWTAuthSpec `json:"jwt,omitempty"`
+}
+
+// APIKeyAuthSpec configures API key authentication.
+type APIKeyAuthSpec struct{}
```

Comment:
Why don't we have a `keySource` configuration

Reply:
We might want this in the future but for now I think it's best to keep the configuration as small as possible. I mirrored what people can configure on the UI and there's no such configuration.

The key source is hardcoded that way:

```golang
KeySource: platform.Source{
    Header:           "Authorization",
    HeaderAuthScheme: "Bearer",
    Query:            "api_key",
},
```